src/main/java/com/couchbase/lite/LiveQuery.java

```diff
@@ -149,93 +117,87 @@ ListenerToken addChangeListener(Executor executor, QueryChangeListener listener)
      * NOTE: this method is synchronized with Query level.
      */
     void removeChangeListener(ListenerToken token) {
+        if (changeNotifier.removeChangeListener(token) <= 0) { stop(); }
+    }
+
+    /**
+     * Starts observing database changes and reports changes in the query result.
+     */
+    void start() {
+        final Database db = query.getDatabase();
+        if (db == null) { throw new IllegalArgumentException("live query database cannot be null."); }
+
         synchronized (lock) {
-            if (changeNotifier.removeChangeListener(token) == 0) { stop(true); }
+            if (state.compareAndSet(State.STOPPED, State.STARTED)) {
+                db.addActiveLiveQuery(this);
+                dbListenerToken = db.addChangeListener(this);
+            }
+
+            // Here if state is STARTED or SCHEDULED:
+            // this method may be called while the LiveQuery is already running,
+            // for example, by Query.setParameters().
+            // If that happens we want to kick off another query, right now
+            releaseResultSetSynchronized();
         }
+
+        update();
```

Comment:
It seems like if adding two listeners in a row (might not be a good use case but coding wise, it's possible), there could be two queries executed instead of just one.

Reply:
I don't think that can happen.  Before `addChangeListener` returns, the LiveQuery is State.SCHEDULED.  No further queries can be scheduled until the first one starts running.